var/spack/repos/builtin/packages/miopen-hip/package.py

```diff
@@ -27,20 +29,36 @@ class MiopenHip(CMakePackage):
     depends_on('bzip2', type='link')
     depends_on('sqlite', type='link')
     depends_on('half', type='build')
-    for ver in ['3.5.0', '3.7.0', '3.8.0']:
+
+    patch('0001-Add-rocm-path-and-rocm-device-lib-path-flags.patch', when='@3.9.0:')
+
+    for ver in ['3.5.0', '3.7.0', '3.8.0', '3.9.0', '3.10.0']:
         depends_on('hip@' + ver, type='build', when='@' + ver)
         depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
         depends_on('comgr@' + ver, type='link', when='@' + ver)
         depends_on('llvm-amdgpu@' + ver, type='build', when='@' + ver)
         depends_on('rocm-clang-ocl@' + ver, type='build', when='@' + ver)
         depends_on('rocblas@' + ver, type='link', when='@' + ver)
         depends_on('rocm-device-libs@' + ver, type='link', when='@' + ver)
+        if ver in ['3.9.0:']:
+            depends_on('zlib', type=('link'))
+
+    def setup_build_environment(self, build_env):
+        if '@3.9.0:' in self.spec:
+            lib_dir = self.spec['zlib'].prefix.lib
+            build_env.prepend_path('LIBRARY_PATH', lib_dir)
+            build_env.prepend_path('LD_LIBRARY_PATH', lib_dir)
```

Comment:
`LD_LIBRARY_PATH` is only used at run-time, this probably isn't needed.

Reply:
You are right here...without setting LD_LIBRARY_PATH being set, the recipe seems to go through fine. Will remove this line